### PR TITLE
Feature/159597052 emissions chart

### PR DESCRIPTION
**Pull request by j8seangel**
Homepage emissions chart
image

Includes:
- [x] Data providers for world bank data, emissions meta and emissions for South Africa from Climate Watch API.
- [x] Dropdown populated and updated by url.
- [x] Modal metadata with all the logic for fetching the data just with the slug of the indicator.

Pending:
- [x] Include loaders and placeholder.
- [x] Calculations per capita and per GDP.
- [x] Button supporting custom links.

### CODE REVIEW

**Comment by aabdaab** [comment]
It's working great! Just few things that I see are not aligned with the design:
- there's a download button missing next to info icon: 
image
- on hover there should be this handy icon: 
image
- line color should be green: 
image
- and the legend is missing: 
image
- as for the legend, should there be something preselected on the chart (as the legend is referring to a dot which is not visible unless you hover on the chart)

**Comment by j8seangel** [comment]
Except for the hover icon the rest of the comments are fixed, thanks for spotting them!

File: app/javascript/app/components/total-ghg-emissions/total-ghg-emissions-styles.scss [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  margin-bottom: $gutter-padding;
+
+  @media #{$tablet-landscape} {
+    margin-top: 200px;
+  }
+  border-bottom: none;
+}
+
+.title {
+  font-family: Lato;
```
Comment (aabdaab) [inline app/javascript/app/components/total-ghg-emissions/total-ghg-emissions-styles.scss]: you can use `$font-family-1` variable defined in settings.scss

File: app/javascript/app/providers/ghg-emissions-provider/ghg-emissions-provider.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import * as actions from './ghg-emissions-provider-actions';
+import reducers, { initialState } from './ghg-emissions-provider-reducers';
+
+class LoginProvider extends PureComponent {
```
Comment (aabdaab) [inline app/javascript/app/providers/ghg-emissions-provider/ghg-emissions-provider.js]: change the name please from `LoginProvider` to `GHGEmissionsProvider` or sth similar, the naming must have stayed here by accident

File: app/javascript/app/providers/ghg-meta-provider/ghg-meta-provider.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import * as actions from './ghg-meta-provider-actions';
+import reducers, { initialState } from './ghg-meta-provider-reducers';
+
+class LoginProvider extends PureComponent {
```
Comment (aabdaab) [inline app/javascript/app/providers/ghg-meta-provider/ghg-meta-provider.js]: ^ as above, the naming of provider

File: app/javascript/app/providers/world-bank-provider/world-bank-provider.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import * as actions from './world-bank-provider-actions';
+import reducers, { initialState } from './world-bank-provider-reducers';
+
+const { COUNTRY_ISO } = process.env;
+
+class LoginProvider extends PureComponent {
```
Comment (aabdaab) [inline app/javascript/app/providers/world-bank-provider/world-bank-provider.js]: ^ naming

File: app/javascript/app/services/api.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+function handleResponse(d) {
+  if (d.status >= 200 && d.status <= 300) {
+    const data = isFunction(d.json) ? d.json() : d;
+    data.json = () => data;
+    return data;
+  }
+  throw new Error(d.statusText);
+}
+
```
Comment (aabdaab) [inline app/javascript/app/services/api.js]: it's just for discussion: why not using `axios` for API calls for example?

Comment (j8seangel) [reply app/javascript/app/services/api.js]: We have been using `fetch` trying to follow the standard but we could discuss it, going to the "concerns" doc

**Pull request merged**